src/Google.Cloud.Functions.Framework.Tests/GcfEvents/GcfConvertersTest.cs

```diff
@@ -166,36 +165,36 @@ public async Task MinimalValidEvent_RawPubSub()
         public async Task InvalidRequest_FirebaseAnalytics(string resourceName)
         {
             var context = GcfEventResources.CreateHttpContext(resourceName);
-            await Assert.ThrowsAsync<GcfConverters.ConversionException>(() => GcfConverters.ConvertGcfEventToCloudEvent(context.Request, s_jsonFormatter));
+            await Assert.ThrowsAsync<GcfConverters.ConversionException>(() => GcfConverters.ConvertGcfEventToCloudEvent(context.Request, s_jsonFormatter, NullLogger.Instance));
         }
 
         [Theory]
-        [InlineData(null, GcfConverters.DefaultRawPubSubTopic)]
-        [InlineData("/not/a/matching/path", GcfConverters.DefaultRawPubSubTopic)]
-        [InlineData("/projects/abc/topics/bcd", "projects/abc/topics/bcd")]
-        public async Task RawPubSubTopic_NoPath(string? path, string expectedTopicResourceName)
+        [InlineData(null, GcfConverters.DefaultRawPubSubTopic, true)]
+        [InlineData("/not/a/matching/path", GcfConverters.DefaultRawPubSubTopic, true)]
+        [InlineData("/projects/abc/topics/bcd", "projects/abc/topics/bcd", false)]
+        public async Task RawPubSubTopic_NoPath(string? path, string expectedTopicResourceName, bool expectWarning)
         {
+            var logger = new MemoryLogger("converter");
             var context = GcfEventResources.CreateHttpContext("raw_pubsub.json", path);
-            var cloudEvent = await GcfConverters.ConvertGcfEventToCloudEvent(context.Request, s_jsonFormatter);
+            var cloudEvent = await GcfConverters.ConvertGcfEventToCloudEvent(context.Request, s_jsonFormatter, logger);
             Assert.Equal($"//pubsub.googleapis.com/{expectedTopicResourceName}", cloudEvent.Source?.ToString());
             string expectedTopic = expectedTopicResourceName.Split('/').Last();
             Assert.Equal(expectedTopic, (string) cloudEvent["topic"]!);
+            Assert.Equal(expectWarning, logger.ListLogEntries().Any(entry => entry.Level >= LogLevel.Warning));
```

Comment:
I'm guessing "unknown topic" is the only thing we are warning on here, right?

Reply:
Yup.